src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/AppointmentsServiceTest.kt

```diff
@@ -209,4 +227,45 @@ internal class AppointmentsServiceTest {
     assertThat(appointments.first().appointmentTime).isEqualTo(actionPlanAppointment.appointmentTime)
     assertThat(appointments.first().durationInMinutes).isEqualTo(actionPlanAppointment.durationInMinutes)
   }
+
+  @Test
+  fun `update appointment with attendance`() {
+    val appointmentId = UUID.randomUUID()
+    val sessionNumber = 1
+    val sessionAttendance = Attended.YES
+    val additionalInformation = "extra info"
+    val createdByUser = SampleData.sampleAuthUser()
+    val actionPlan = SampleData.sampleActionPlan()
+
+    val existingAppointment = SampleData.sampleActionPlanAppointment(actionPlan = actionPlan, createdBy = createdByUser)
+
+    whenever(actionPlanAppointmentRepository.findByActionPlanIdAndSessionNumber(appointmentId, sessionNumber))
+      .thenReturn(existingAppointment)
+    whenever(actionPlanAppointmentRepository.save(any())).thenReturn(existingAppointment)
+
+    val savedAppointment = appointmentsService.recordAttendance(appointmentId, 1, sessionAttendance, additionalInformation)
+    val argumentCaptor: ArgumentCaptor<ActionPlanAppointment> = ArgumentCaptor.forClass(ActionPlanAppointment::class.java)
+
+    verify(actionPlanAppointmentRepository).save(argumentCaptor.capture())
+    assertThat(argumentCaptor.firstValue.attended).isEqualTo(sessionAttendance)
+    assertThat(argumentCaptor.firstValue.additionalAttendanceInformation).isEqualTo(additionalInformation)
+    assertThat(argumentCaptor.firstValue.attendanceSubmittedAt).isNotNull
```

Comment:
💭 Should we assert here that the submission time corresponds to the current time?